lib/io.toit

```diff
@@ -1545,7 +1587,7 @@ class EndianBuffer extends EndianWriter:
   Writes the given signed int16 $value to this buffer at the given index $at.
   */
   put-int16 --at/int value/int:
-    endian_.put-int16 cached-byte-array_ at value
+    byte-order_.put-int16 cached-byte-array_ at value
```

Comment:
Isn't this completely broken? Shouldn't you always write to the cached array at index 0?

Reply:
Good catch.
Fixed and added a test.